CitationStyleLanguagePlugin.php

```diff
@@ -583,15 +583,23 @@ public function getCitation(PKPRequest $request, Submission $submission, string
                     $additionalMarkup = [
                         'DOI' => [
                             'function' => function ($item, $renderedValue) {
-                                return '<a href="https://doi.org/' . $item->DOI . '">' . $renderedValue . '</a>';
+                                $prefixPattern = '/.*DOI:/i';
+                                if (preg_match($prefixPattern, $renderedValue, $prefixMatches)) {
+                                    $renderedValue = preg_replace($prefixPattern, '', $renderedValue);
+                                }
+                                $suffixPattern = '#'.$item->DOI.'(.*)$#';
+                                if (preg_match($suffixPattern, $renderedValue, $suffixMatches)) {
+                                    $renderedValue = str_replace($suffixMatches[1], '', $renderedValue);
+                                }
+                                return $prefixMatches[0] . '<a href="https://doi.org/' . $item->DOI . '">' . $renderedValue . '</a>' . $suffixMatches[1];
```

Comment:
Perhaps this would be ok:
```php
$itemDoi = '10.1016/j.jpsychires.2017.11.014';
$renderedValue = "[Prefix data] $itemDoi [Suffix data]";
str_replace($itemDoi, "<a href=\"https://doi.org/{$itemDoi}\">{$itemDoi}</a>", $renderedValue);
```

But if you really need to validate if there's a `DOI:` on the beginning, you can collect the prefix + DOI + suffix, then validate if it has the `DOI` text using the `preg_replace_callback()`.

Reply:
So maybe I could only search after something like `(https://doi.org/)*$itemDoi` to replace it...